### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by marcinadd**
Device registration prototype

### CODE REVIEW

**Review by ninjavaz — APPROVED** [review]

**Review by ninjavaz — APPROVED** [review]

**Review by ninjavaz — APPROVED** [review]

**Review by BartlomiejRasztabiga — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,29 @@
...
+# Installation
+1. Install required dependencies:
+```shell script
+   $ pip install -r requirements.txt
+```
+1. Build the library (you will get **.whl** in **dist** directory):
+```shell script
+   $ python setup.py bdist_wheel
+```
+1. Now you can install it via pip:
```
Comment (thewildtree) [inline README.md:13]: the numbering here seems to have gotten a little bit stuck at 1

File: thingooConnector/connector.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import requests
+
+from config import REGISTER_ENDPOINT
+
+logger = logging.getLogger(__name__)
+
+
+class ClientCredentials:
+    """
+    A class used for store client credentials which are used to get access token
```
Comment (thewildtree) [inline thingooConnector/connector.py]:
```suggestion
    A class used to store client credentials required to retrieve the access token.
```

File: thingooConnector/connector.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    def client_secret(self):
+        return self._client_secret
+
+
+class HTTPException(Exception):
+    def __init__(self, error_text, status_code, text):
+        super().__init__(f'{error_text}: {status_code} {text}')
+
+
+class RetrieveTokenException(HTTPException):
```
Comment (thewildtree) [inline thingooConnector/connector.py]:
```suggestion
class TokenRetrievalException(HTTPException):
```
\+ usages

File: thingooConnector/connector.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    def __init__(self, status_code, text):
+        super().__init__("Fail to retrieve access token", status_code, text)
```
Comment (thewildtree) [inline thingooConnector/connector.py]:
```suggestion
        super().__init__("Failed to retrieve access token", status_code, text)
```

Comment (thewildtree) [reply thingooConnector/connector.py]: (same thing below)

File: thingooConnector/connector.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+
+class RegisterDeviceException(HTTPException):
```
Comment (thewildtree) [inline thingooConnector/connector.py]:
```suggestion
class DeviceRegistrationException(HTTPException):
```

File: thingooConnector/connector.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    def __init__(self, status_code, text):
+        super().__init__("Fail to register device", status_code, text)
+
+
+class Token:
+    """
+    A class used for store token
```
Comment (thewildtree) [inline thingooConnector/connector.py]:
```suggestion
    A class used to store the access token.
```

File: thingooConnector/connector.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    """
+
+    def __init__(self, json):
+        """
+        Create token from json
```
Comment (thewildtree) [inline thingooConnector/connector.py]:
```suggestion
        Creates a token instance from the provided JSON.
```

File: thingooConnector/connector.py [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        self._access_token = json["access_token"]
+        self._expires_in = json["expires_in"]
+
+    def access_token(self):
+        return self._access_token
+
+
+class Connector:
+    """
+    Connector class to Thingoo instance
```
Comment (thewildtree) [inline thingooConnector/connector.py]:
```suggestion
    A class used for connecting the device to a chosen Thingoo instance.
```

**Review by thewildtree — COMMENTED** [review]
just wording nitpicks for now; looks good code-wise at a first glance, i'll try to test it later and do a proper code review if needed.

**Pull request merged**